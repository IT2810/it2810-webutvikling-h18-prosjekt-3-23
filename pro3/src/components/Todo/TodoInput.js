import React, { Component } from 'react';
import { TextInput } from 'react-native';

export default class TodoInput extends Component {

    render() {
        return (
            <TextInput
                    style={this.props.styleTextInput}
                    onChangeText={this.props.changeTextHandler}
                    onSubmitEditing={this.props.addTask}
                    value={this.props.text}
                    placeholder="+ Add a new task"
                    returnKeyType="done"
                    returnKeyLabel="done"
                    placeholderTextColor="rgba(96, 125, 139, 0.631)"
            />
        )
    }
}