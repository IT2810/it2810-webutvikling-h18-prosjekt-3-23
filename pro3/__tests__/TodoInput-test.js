import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TodoInput from '../src/components/Todo/TodoInput';

it('renders correctly', () => {
const tree = renderer.create(<TodoInput/>).toJSON();
expect(tree).toMatchSnapshot();
spyOn(console, 'error');
});
