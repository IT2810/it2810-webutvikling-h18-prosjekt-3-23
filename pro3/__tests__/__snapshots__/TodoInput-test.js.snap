// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  width="100%"
>
  <TextInput
    allowFontScaling={true}
    placeholder="+ Add a new todo"
    placeholderTextColor="rgba(96, 125, 139, 0.731)"
    returnKeyLabel="done"
    returnKeyType="done"
    underlineColorAndroid="transparent"
  />
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       Click on 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       when task is completed
    </Text>
  </View>
</View>
`;
